include/pybind11/cast.h

```diff
@@ -74,6 +75,19 @@ PYBIND11_NOINLINE inline internals &get_internals() {
     return *internals_ptr;
 }
 
+// Walks up the inheritance tree of `derived` until a registered (python) type is found, and returns
+// that base class type_info.  Returns nullptr if no registered base class is found.
+PYBIND11_NOINLINE inline detail::type_info* get_parent_type_info(PyTypeObject *type) {
```

Comment:
Indeed, it's a leftover function from a previous implementation detail.  Removed.